monitoring/sdm_health_exporter/sdm_health_exporter.py

```diff
@@ -0,0 +1,178 @@
+    Returns: dictionary of SdmObject instances, keyed by resource name
+    """
+
+    # prometheus labels that will be attached to the exported metric
+    labels = ['id','name','tags','task']
+
+    for resource in client.resources.list(''):
+        
+        # make sure there are no characters that are invalid for the prometheus exporter
+        resource_name = strip_invalid_chars(resource.name)
+
+        # only process resources that are tagged with <alert_tag> in strongDM
+        if alert_tag not in resource.tags:
+            continue
+
+        # only register a new prometheus collector if it's not an existing object
+        # create a new "Gauge" and define the labels that will be used for the metric
+        # create new SdmObject with this information and add it to "sdm_objects" dictionary
+        if resource_name not in sdm_objects:
+            health_metric = Gauge(resource_name, 'health of resource', labelnames=labels)
+            sdm_objects[resource_name] = SdmObject(health_metric=health_metric)
+
+        # resource health is returned as True or False
+        # set the label values based on information retrieved from the API call
+        # set the metric to 0 (healthy) or 1 (unhealthy)
+        sdm_objects[resource_name].health_metric.labels(
+            id=resource.id,
+            name=resource_name,                       
+            tags=resource.tags,
+            task='health_check'
+        ).set(0 if resource.healthy else 1)
+
+    return sdm_objects
+
+
+def get_sdm_nodes(client, alert_tag, sdm_objects):
+    """
+    Issue strongDM API calls every <update_interval> to collect information on nodes
+    Export any resource that's tagged with <alert_tag> in strongDM to a prometheus exporter
+    Returns: dictionary of SdmObject instances, keyed by node name
+    """
+
+    # prometheus labels that will be attached to the exported metric
+    labels = ['id','name','tags','task']
+
+    for node in client.nodes.list(''):
+        
+        # make sure there are no characters that are invalid for the prometheus exporter
+        node_name = strip_invalid_chars(node.name)
+
+        # only process nodes that are tagged with <alert_tag> in strongDM
+        if alert_tag not in node.tags:
+            continue
+
+        # only register a new prometheus collector if it's not an existing object
+        # create a new "Gauge" and define the labels that will be used for the metric
+        # create new SdmObject with this information and add it to "sdm_objects" dictionary
+        if node_name not in sdm_objects:
+            health_metric = Gauge(node_name, 'health of resource', labelnames=labels)
+            sdm_objects[node_name] = SdmObject(health_metric=health_metric)
+
+        # node health is returned as "started" or "stopped"
+        # make sure that the "state" value we received is expected
+        if node.state not in ("started", "stopped"):
```

Comment:
We don't really care about any other value than `started`, so not sure if we want to keep this validation. In case you really want to keep it, we need to include `new`:

```
if node.state not in ("started", "stopped", "new"):
```